docs/src/app/components/pages/customization/themes.md

```diff
@@ -0,0 +1,232 @@
+import React from 'react';
+import Colors from 'material-ui/lib/styles/colors';
+import MuiThemeProvider from 'material-ui/lib/MuiThemeProvider';
+import getMuiTheme from 'material-ui/lib/styles/getMuiTheme';
+import AppBar from 'material-ui/lib/app-bar';
+
+const muiTheme = getMuiTheme({palette: {textColor: Colors.cyan500}});
+
+class Main extends React.Component {
+  render() {
+    return (
+      <MuiThemeProvider muiTheme={muiTheme}>
+        <AppBar title="Hello World!"/>
+      </MuiThemeProvider>
+    );
+  }
+}
+
+export default Main;
+```
+
+Internally, Material-UI components use React's context feature to implement theming. Context is a way
+to pass down values through the component hierarchy without having to use props at every level.
+In fact, context is very convenient for concepts like theming, which are usually implemented in
+a hierarchical manner.
+
+`MuiThemeProvider` uses this feature to pass down the theme to components that need it.
+
+## Using the theme
+
+In case you wish to access the theme object yourself you can use the `muiThemeable` decorator:
+
+```js
+import React from 'react';
+import muiThemeable from 'material-ui/lib/muiThemeable';
+
+class DeepDownTheTree extends React.Component {
+
+  static propTypes = {
+    muiTheme: React.PropTypes.object,
+  };
+
+  render() {
+    return (
+      <span style={{color: this.props.muiTheme.baseTheme.palette.textColor}}>
+        Hello World!
+      </span>
+    );
+  }
+}
+
+export default muiThemeable()(DeepDownTheTree);
+```
+
+`muiThemeable` gets the theme from context and passes it down as a property.
+
+## Predefined themes
+
+We ship two base themes with Material-UI: light and dark. They are located
+under `material-ui/lib/styles/baseThemes/`. Custom themes may be defined similarly.
+
+The `lightBaseTheme` is the default so you will not need to do anything to use it.
+But for the `darkBaseTheme` you can use this snippet:
+
+```js
+import React from 'react';
+import darkBaseTheme from 'material-ui/lib/styles/baseThemes/darkBaseTheme';
+import MuiThemeProvider from 'material-ui/lib/MuiThemeProvider';
+import getMuiTheme from 'material-ui/lib/styles/getMuiTheme';
+import AppBar from 'material-ui/lib/app-bar';
+
+const darkMuiTheme = getMuiTheme(darkBaseTheme);
+
+class Main extends React.Component {
+  render() {
+    return (
+      <MuiThemeProvider muiTheme={darkMuiTheme}>
+        <AppBar title="Hello World!"/>
+      </MuiThemeProvider>
+    );
+  }
+}
+
+export default Main;
+```
+
+## Using context
+
+The `MuiThemeProvider` and `muiThemeable` simply use context. If you prefer
+using context instead of these you can follow these pattern:
+
+Pass theme down the context:
+
+```js
+import React from 'react';
+import darkBaseTheme from 'material-ui/lib/styles/baseThemes/darkBaseTheme';
+import getMuiTheme from 'material-ui/lib/styles/getMuiTheme';
+import AppBar from 'material-ui/lib/app-bar';
+
+class Main extends React.Component {
+
+  static childContextTypes = {
```

Comment:
This is more of a general discussion, I'm fine with this for now as is. But someone recently on gitter requested that we don't use the static properties/static property initializers in example code because it requires a lower babel stage (I think 1). What do you think?